data/darktableconfig.xml.in

```diff
@@ -281,10 +281,17 @@
   </dtconfig>
   <dtconfig prefs="storage" section="xmp">
     <name>write_sidecar_files</name>
-    <type>bool</type>
-    <default>true</default>
+    <type>
+      <enum>
+        <option>never</option>
+        <option>very lazy</option>
```

Comment:
"lazy" is a term used often in computing... and that "look up in manual" is just chef's kiss ;)

Reply:
Sure lazy is a common terms in computer science, but if we could convey at least a bit of semantics via the naming to help remembering what is the difference between "lazy" and "very lazy" it would be best to me :)

Reply:
It'd be hard to convey any more meaning than `lazy` & `very lazy` convey without making it too verbose. I mean do

- never
- on non-automated edit
- on process
- always

convey more meaning than lazy and very lazy?

Reply:
i thought i already commented via mobile, i have thought about this and couldn't find a better wording; wouldn't even be sure what to do better in german. "after basic" / "after presets" came to my mind but that would probably lead to more questions. maybe one of the native english speaking people as @elstoc having worked so much on docs & wording might have a better idea?

Reply:
Maybe, but I haven't paid attention and I'm still not 100% sure I know exactly how these modes work. Perhaps if you could give a very detailed description (like what you would expect to see if someone looks it up in the manual) and I might be able to work from that.

Reply:
Here's what it does:
both lazy & very lazy will write xmp whenever user labels, tags, geotags or edits metadata.
difference is in interpreting edits done to photo:
- lazy will write xmp if any edit/preset was applied including auto presets - so eg if user autoapplies lens correction the xmp will be written.
- very lazy will write xmp only if edit done wasn't "auto" meaning xmp won't be written if autoapplied preset for lens correction, but will be written if you tweak ANYTHING or paste edits from anotehr file or apply style etc.

Reply:
Ok. I understand "never" but how does "lazy" differ from "always"

Reply:
always: you just import the image and xmp is written
lazy: you just import the image and xmp is not written

probably single most requested feature ;)

Reply:
I might go with the following (option in bold followed by description, just to confirm I have it right)

- **immediate**: xmp file is written on import and updated on every change to the image
- **on any edit**: xmp file is not written until a change is made to the image, whether manually or by the automatic enabling of a module
- **on manual edit**: xmp file is not written until a change is explicitly made by a user (excluding automatic enabling of a module)
- **never**: xmp file is never written

If this works the way I think it does, we will have a scenario where the edit history in the database and xmp file are out of synch (until a user touches a slider). If I've understood correctly, I'm not sure I'm comfortable with that. Perhaps if we're not writing history to the xmp, we shouldn't write to the database history either. The auto modules will reapply next time the user opens the image anyway.

I usually cull using the jpeg preview and occasionally pop into the darkroom just to examine an image more closely. To get the jpeg preview back again, I have to discard history, and it might be nice not to have to do that.

Reply:
From your latest comments i think you don't understand exactly how this works / should work.

What i want to achieve is, we don't need to write sidecars until there is at least one step added to history after include into the database. Or in other words, if i didn't do anything on the image i don't have to keep additional data in a sidecar at all as i can always import it again without problems (One exception here, if the presets change ... but that would be user responsability)

So in "very lazy" mode, you include a bunch of images, auto presets might be applied and you can view any image, switch between them or whatever you want and no sidecar won't we written as there is no need to do so. So not "out of synch" but just no sidecar.

**But** if you add a module, remove it or change any parameter (hash is changed) or add ad tag, metadata ... we **will** write sidecars from now on exactly as it has always been done. That's why i named these mosed "lazy" and not "late" or alike. From the user side it means, sidecars to be used for reimporting or archiving need to be written (and are written) **only** after any sort of editing.

In "lazy" mode we will write sidecars if any change takes place after **any** change has been done after import & applying mandatory modules.

"immediate" does not sound better than "always" for me because it would be versus "late" or "later" and this is not about timing but about a condition.

Reply:
Hmm I think I have understood it but maybe my definitions could have been clearer. Maybe "out of synch" was the wrong thing to say but still you would normally expect the xmp to reflect the development history in the database and here it won't (because there is no xmp). Lazy is still the wrong term though. Lazy implies "I will do it but later when I have some time" but here we're actively not doing it until a certain condition has been reached, after which it's the same as "always".

So perhaps the option should be save edits to XMP files **after import**, **after first edit**, **after first manual edit** or **never**.

Personally I think I would probably not want to use the third option for the reasons I've stated.

Reply:
Ok, for me your suggestions sound better and give a better idea of use case.

Reply:
> Personally I think I would probably not want to use the third option for the reasons I've stated

I think i understand your point, it's a matter of personal preference. I myself would only want "real edits" kept in sidecars - if i remove an image from the database for backup and later re-import to make use of it i would expect updated presets to be better than they were last time so less work to be done.